### PR TITLE
Feature/school data cleanup debugging

### PR DESCRIPTION
**Pull request by hagenhaeussler**
What this PR does:

This pull request implements command-line school data cleanup scripts for missing or invalid school countries and inaccurate school grade levels, along with automated RSpec and Cucumber coverage for those workflows.

This debugging follow-up also includes the cleanup needed to get the new files aligned with the repository’s linting rules:

    adds the missing frozen string literal comments
    updates Ruby hash syntax to the preferred shorthand style
    fixes minor formatting and indentation issues flagged by RuboCop

These fixes do not change the intended behavior of the cleanup tasks; they make the branch pass linting more cleanly and keep the new test/support code consistent with the rest of the codebase.

Who authored this PR?

@hagenhaeussler
How should this PR be tested?

    There is no deploy for this change. The main behavior is exercised through the rake tasks and their automated tests.

    Run the RSpec coverage for the rake tasks to verify the script logic around country cleanup and grade-level cleanup.

    Run the Cucumber coverage for the school data cleanup workflow to verify the end-to-end task behavior from test setup through persisted updates.

    Run RuboCop to confirm the new rake task, specs, and step definitions satisfy the repository’s linting requirements.

    For country cleanup, verify that schools with missing or invalid countries are corrected when the script can infer a country from:
        a valid US state abbreviation
        a non-generic website TLD
        reverse geocoding from coordinates when available

    Also verify that the country script leaves records unchanged when:
        the country is already valid
        no heuristic can confidently determine the correct country
        the task is run in dry-run mode

    For grade-level cleanup, verify that the script flags mismatches between a school's grade level and:
        school-name keywords such as "High School", "Middle School", "Elementary", "College", or "University"
        the majority education level of teachers linked to that school

    Verify that high-confidence grade-level matches can be applied, while low-confidence cases are surfaced for manual review instead of being automatically changed.

    Edge cases to watch:
        generic website domains such as .com, .org, or .edu should not be used to infer country
        malformed website URLs should not crash the task
        schools with no usable teacher education-level data should not be auto-corrected from teacher-based logic
        dry-run behavior should report proposed fixes without writing to the database
        reverse geocoding depends on BACKEND_MAPS_API_KEY being present

Are there any complications to deploying this?

No schema changes or data migrations are included.

cc @armandofox @cycomachead @mdawn65 @ronikriger @kienthuynh @NishK05

### CODE REVIEW

File: features/step_definitions/school_data_cleanup_steps.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  unless Rake::Task.task_defined?(task_name)
+    Rake.application.rake_require(
+      "tasks/school_data_cleanup",
+      [Rails.root.join("lib").to_s]
+    )
+  end
+
+  ENV["APPLY"] = "true"
+
+  Rake::Task[task_name].invoke
```
Comment (Ethan-Stone1) [inline features/step_definitions/school_data_cleanup_steps.rb:64]: Would it not be safer to reenable before you invoke? Also if the task fails then ENV.delete("APPLY") is never run

**Comment by zhouyijun111** [comment]
This PR LGTM with clear test coverage and thoughtful handling of edge cases. It could also be helpful to include a screenshot or sample output to illustrate what kinds of changes are being made.

**Comment by kienthuynh** [comment]
This looks sound overall to me. There are some considerations that Ethan made in your code that might be worth checking out, but since this is a single use script it should be fine. 

Make a PR to Michael's golden repo so he can test it out on some dummy data and give any feedback. I'd leave this branch unmerged into our repo for now until Michael confirms its functionality.
